bundles/org.openhab.ui.basic/src/main/java/org/openhab/ui/basic/internal/render/AbstractWidgetRenderer.java

```diff
@@ -423,9 +418,8 @@ protected String processColor(Widget w, String originalSnippet) {
     }
 
     protected @Nullable String getRGBHexCodeFromItemState(@Nullable State itemState) {
-        if (itemState instanceof HSBType) {
-            HSBType hsbState = (HSBType) itemState;
-            return "#" + Integer.toHexString(hsbState.getRGB()).substring(2);
+        if (itemState instanceof HSBType hsbState) {
+            return "#" + Integer.toHexString(ColorUtil.hsbTosRgb(hsbState)).substring(2);
```

Comment:
I agree that your change will change nothing but I am asking myself since when hsbState.getRGB() is returning a sRGB rather a RGB ? Was it changed recently ? I am not sure it does not break the color picker in Basic UI.

Reply:
Change was done in April 2023 but I think it is ok and the result is the same as before.